gui-lib/framework/private/text-port.rkt

```diff
@@ -590,7 +593,25 @@
                         snip))
                    (else
                     (markup->snip (image-markup-alt-markup markup) style #f)))))
-             
+
+             (define (number-markup->string/snip markup)
+               (define number (number-markup-number markup))
+               (define prefix? (number-markup-prefix? markup))
+               (define fraction-view (number-markup-fraction-view markup))
+               (cond
+                 [(inexact? number)
+                  (string-append (if prefix? "#i" "") (number->string number))]
+                 [(integer? number)
+                  (string-append (if prefix? "#e" "") (number->string number))]
+                 [else
+                  (case fraction-view
+                    [(mixed improper)
+                     (define snip (number-snip:make-fraction-snip number prefix?))
+                     (send snip set-fraction-view fraction-view)
+                     snip]
+                    [else ; decimal
+                     (number-snip:make-repeating-decimal-snip number prefix?)])]))
+                   
```

Comment:
Is this code a copy of some code elsewhere?

Reply:
> Is this code a copy of some code elsewhere?

No. I took inspiration from here:

https://github.com/racket/drracket/blob/917c6abdd3b146cfd85b08fdc7ad39f48cd834e3/drracket/drracket/private/language.rkt#L433

(Sorry, did not see this earlier.)